ICAFIX/scripts/melodicicasso.m

```diff
@@ -0,0 +1,348 @@
+      if ~stat
+        fprintf('finished melodic %.3i/%.3i with %.3i steps. ',iI,nI,nSteps(iI));toc
+      else
+        error(' melodic %.3i/%.3i failed!\n\n%s',iI,nI,out)
+      end
+      A{iI} = load([meloDir '/melodic_mix'],'-ascii');
+      W{iI} = pinv(A{iI});
+    end %parfor iI 
+  else
+    fprintf('Level %i: runing %i melodics as single threaded serial loop with %i cores:\n',iL,nI)
+    for iI = 1:nI
+      meloDir = sprintf('%s/%i',tmpDir,iI);
+      if exist(meloDir,'dir');rmdir(meloDir,'s');end
+      mkdir(meloDir);
+      bootFile = sprintf('%s/melodic_boot_vnts',tmpDir);
+      if ~strcmp(bootMode,'randinit')
+        bootFile = sprintf('%s/melodic_boot_vnts',meloDir);
+        X = single(vnts(bootIdx(:,iI),:));
+        X = unmaskAndSpatiallyInflate(X,imSz,brainMask,mtxDim);
+        writeNIFTI(X,bootFile,hdr);
+      end
+      if iL == 1
+        if strcmp(bootMode,'boot')
+          seed = 1;% use the same initialization for each run
+        else
+          seed = iI + seedOffset;
+        end
+        cmd = sprintf(...
+          'melodic -i %s -o %s --nobet --vn --dim="%i" --no_mm --seed="%i" -m %s -v --debug',...
+            bootFile,meloDir,Dim,seed,brainMaskFile);
+      else
+        cmd = sprintf(...
+          'melodic -i %s -o %s --nobet --vn --dim="%i" --no_mm --init_ica="%s" -m %s -v --debug',...
+            bootFile,meloDir,Dim,init_ica,brainMaskFile);
+      end
+      
+      tic;[stat,out] = system(cmd);
+      nSteps(iI) = cellfun(@str2num,(regexp(fileread([meloDir '/log.txt']),'(?<=after)(.*?)(?=steps)','match')));
+      if ~stat
+        fprintf('finished melodic %.3i/%.3i with %.3i steps. ',iI,nI,nSteps(iI));toc
+      else
+        error(' melodic %.3i/%.3i failed!\n\n%s',iI,nI,out)
+      end
+      A{iI} = load([meloDir '/melodic_mix'],'-ascii');
+      W{iI} = pinv(A{iI});
+    end % for iI 
+  end
+
+  fprintf('level %i: %3.1f steps on average\n',iL,mean(nSteps));
+  if iL == 1
+    % run melodic once on original non-resampled data just to get whitening matix
+    % (there doesn't seem to be a way to turn off ica, so using a big epsilon is the best I can do)
+    fprintf('calculating whitening/dewhitening matrices ...\n')
+    meloDir = sprintf('%s/white',tmpDir);
+    mkdir(meloDir);
+    cmd = sprintf(...
+            'melodic -i %s -o %s --Owhite --nobet --vn --dim="%i" --no_mm --eps=0.01 -m %s -v --debug',...
+          vntsFile,meloDir,Dim,brainMaskFile);
+    [stat,out] = system(cmd);
+    if stat;error(' melodic whitening failed!\n\n%s',out);end
+    whiteningMatrix = load([meloDir '/melodic_white'],'-ascii');
+
+    % populate common sR
+    sR = struct();
+    for iiL = nL:-1:1
+      sR(iiL).signal = vnts';
+      sR(iiL).whiteningMatrix = whiteningMatrix;
+      sR(iiL).dewhiteningMatrix = pinv(whiteningMatrix);
+
+    end
+  end %if iL == 1
+
+  % populate level-specific sR
+  sR(iL).index = sortrows(combvec(1:nI,1:Dim)');%flip(combvec(1:Dim,1:nI)',2)
```

Comment:
This also needs to have some workaround, it seems like `combvec` somehow manages to tell the compiler that parallel needs to be included (and thus, a 200MB binary is produced).